config/configmodels/configmodels.go

```diff
@@ -43,12 +47,31 @@ type Config struct {
 // Type is the component type as it is used in the config.
 type Type string
 
+// String returns the type as a string.
+func (typ Type) String() string {
+	return string(typ)
+}
+
+// FullName is the full identifier (e.g. receiver or receiver/1).
+type FullName string
```

Comment:
Should this be a proper struct with two internal fields type and name?

Reply:
I don't have a strong preference either way. I had it as a struct before but I believe @tigrannajaryan preferred the string (please correct me if I'm wrong).

Reply:
@jrcamp I think what you have is ok but if @bogdandrutu has a strong opinion on making it a struct I am fine with that approach too.

Reply:
@bogdandrutu what do you think? 99% of the time we're using the full name so if we make it a struct that decomposes into a type and name it'd be have to either concat on each usage or just store it inside the struct pre-concated.

Reply:
The fields are internal so we can store "type" "name" and "fullName" nothing stops us to do this optimization. What I am trying to say is:
1. It should be a struct not a `typedef` (C++ ftw)
2. Expos functions like "Type()" "Name()" "String()" (or "FullName")

Reply:
The reason of not having a typedef is that parsing the internal fields may cause errors, and by having the validation happening only in one place that simplifies the logic. You can keep it as typedef if you want but then the ctor must validate the parsing in order to have the helper functions not returning an error.